src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -216,6 +218,40 @@ public void waitForElementToDisappear(By by){
     }
     
     /**
+     * Waits for a list of elements to be invisible or not present, or timeout.
+     */
+    public void waitForElementsToDisappear(List<WebElement> elements) {
+        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.inst().TEST_TIMEOUT);
+        wait.until(invisibilityOfAllElements(elements));
+    }
+    
+    /**
+     * Code adapted from SeleniumHQ's GitHub page.
+     * TODO to be removed when Selenium is upgraded to the version supporting this method.
+     *
+     * An expectation for checking all elements from given list to be invisible
+     *
+     * @param elements used to check their invisibility
+     * @return Boolean true when all elements are not visible anymore
+     */
+    private ExpectedCondition<Boolean> invisibilityOfAllElements(final List<WebElement> elements) {
+        return new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver webDriver) {
+                for (WebElement element : elements) {
+                    try {
+                        if (element.isDisplayed()) {
+                            return false;
+                        }
+                    } catch (Exception e) {
```

Comment:
Nope. The code here is [copy-pasted verbatim](https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/support/ui/ExpectedConditions.java#L1384) (+ style fixes).